acceptance/indy_acceptance/utilities/utils.py

```diff
@@ -0,0 +1,161 @@
+"""
+Created on Nov 9, 2017
+
+@author: khoi.ngo
+
+Containing all functions used by several test steps on test scenarios.
+"""
+from indy.error import IndyError
+from .constant import Color
+from . import constant
+from .result import Status
+
+
+def generate_random_string(prefix="", suffix="", size=20):
+    """
+    Generate random string .
+
+    :param prefix: (optional) Prefix of a string.
+    :param suffix: (optional) Suffix of a string.
+    :param size: (optional) Max length of a string (include prefix and suffix)
+    :return: The random string.
+    """
+    import random
+    import string
+    left_size = size - len(prefix) - len(suffix)
+    random_str = ""
+    if left_size > 0:
+        random_str = ''.join(random.choice(
+            string.ascii_uppercase + string.digits) for _ in range(left_size))
+    else:
+        print("Warning: Length of prefix and suffix more than %s chars"
+              % str(size))
+    result = str(prefix) + random_str + str(suffix)
+    return result
+
+
+def exit_if_exception(result):
+    """
+    If "result" is an exception then raise the "result".
+    Unless "result" is an exception then return the "result".
+    :param result: the value that you want to check.
+    :return: "result" if it is not an exception.
+    """
+    if (isinstance(result, Exception)):
+        exit(1)
```

Comment:
It will be better just to re-raise the exception 'result' rather than to abort with `exit` call.

By the way, `exit` internally raises `SystemExit` which is inherited from `BaseException`. Thus with `exit` we will not enter to `except` section of `try` block in `TestScenarioBase.execute_scenario`. Anyway, `exit` is some kind of emergency ways to quit. It doesn't fit an ordinary test fail.

Reply:
Fyi, the execution was terminated only after setting the test's status as failed.

Updated as "raise_if_exception". The new raised exception will be handled at TestScenarioBase,.execute_scenario.